docs/StardustDocs/topics/predicateJoin.md

```diff
@@ -0,0 +1,283 @@
+[//]: # (title: predicateJoin)
+
+<!---IMPORT org.jetbrains.kotlinx.dataframe.samples.api.Join-->
+
+Joins two [`DataFrames`](DataFrame.md) by a join expression. 
+
+```kotlin
+predicateJoin(otherDf, type = JoinType.Inner) [ { joinExpression } ]
+
+joinExpression: JoinedDataRow.(JoinedDataRow) -> Boolean
+
+interface JoinedDataRow: LeftDataRow {
+    
+    val right: RightDataRow
+    
+}
+```
+
+Unlike [join](join.md) similar to one from relational databases, where relations between tables are based on IDs and usually equality between values, predicate join allows any operations that returns Boolean value. That gives you the ability to join data based on various logical operations and heuristics. This can also be useful when dealing with data from disparate sources that are connected, but lack common identifiers or structured relations typical of SQL tables with indices.
```

Comment:
please fold the line so I don't have to scroll horizontally to read it, haha

Reply:
i rewrote it completely in the end

Reply:
much better :)